sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch_app_sdk/inject/SageResearchAppSDKModule.java

```diff
@@ -1,42 +1,83 @@
+import org.sagebionetworks.research.sageresearch.dao.room.ScheduledActivityEntityDao;
+import org.sagebionetworks.research.sageresearch.viewmodel.ScheduleRepository;
 import org.sagebionetworks.research.sageresearch.viewmodel.ScheduledActivityTaskResultProcessor;
+import org.sagebionetworks.research.sageresearch.viewmodel.ScheduledRepositorySyncStateDao;
 import org.sagebionetworks.research.sageresearch_app_sdk.TaskResultUploader;
 import org.sagebionetworks.research.sageresearch_app_sdk.archive.AbstractResultArchiveFactory;
 import org.sagebionetworks.research.sageresearch_app_sdk.archive.AbstractResultArchiveFactory.ResultArchiveFactory;
 import org.sagebionetworks.research.sageresearch_app_sdk.archive.SageResearchResultArchiveFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
 
-import java.util.Arrays;
-import java.util.List;
+import javax.inject.Singleton;
 
 import dagger.Binds;
 import dagger.Module;
 import dagger.Provides;
+import dagger.multibindings.ElementsIntoSet;
 import dagger.multibindings.IntoSet;
 
-@Module
+@Module(includes = {})
 public abstract class SageResearchAppSDKModule {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SageResearchAppSDKModule.class);
+
+    private static final String RESEARCH_DB_FILENAME = "org.sagebionetworks.research.ResearchDatabase";
+
     @Provides
     static AbstractResultArchiveFactory provideAbstractResultArchiveFactory(
             ImmutableList<ResultArchiveFactory> resultArchiveFactories) {
         return new SageResearchResultArchiveFactory(resultArchiveFactories);
     }
 
-    @IntoSet
-    @Binds
-    abstract TaskResultProcessor provideScheduledActivityTaskResultProcessor(
-            ScheduledActivityTaskResultProcessor taskResultProcessor);
-
     /**
      * @return list of processors to receive final task results
      */
     @Provides
-    static List<TaskResultProcessor> provideTaskResultProcessors(
+    @ElementsIntoSet
+    @BridgeApplicationScope
+    static Set<TaskResultProcessor> provideTaskResultProcessors(
             TaskResultUploader taskResultUploader,
             ScheduledActivityTaskResultProcessor scheduledActivityTaskResultProcessor) {
-        return Arrays.asList(taskResultUploader, scheduledActivityTaskResultProcessor);
+        LOGGER.debug("Providing TaskResultProcessors");
+
+        return Sets.newHashSet(taskResultUploader, scheduledActivityTaskResultProcessor);
+    }
+
+    @Provides
+    @BridgeApplicationScope
+    static ResearchDatabase provideResearchDatabase(Context context) {
+        LOGGER.debug("Providing ResearchDatabase");
+        return Room.databaseBuilder(context, ResearchDatabase.class, RESEARCH_DB_FILENAME)
+                .build();
+    }
+
+    @Provides
+    static ScheduledActivityEntityDao provideScheduledActivityDao(ResearchDatabase researchDatabase) {
+        return researchDatabase.scheduleDao();
+    }
+
+    @Provides
+    @BridgeApplicationScope

```

Comment:
Ahh gotcha, so the provider is the one that specifies the scope.

Reply:
That, or classes with an @Inject constructor can specify their scope, but I found that less flexible